cmd/device/rm/rm.go

```diff
@@ -38,22 +41,40 @@ You can use: '$ edgex device list' to find a device's name and ID.`,
 			}
 
 			deviceID := args[0]
-			respBody, err := client.DeleteItem(deviceID,
-				config.Conf.MetadataService.DeviceByIDRoute,
-				config.Conf.MetadataService.DeviceBySlugNameRoute,
-				config.Conf.MetadataService.Port)
+
+			ctx, _ := context.WithCancel(context.Background())
+
+			url := config.Conf.MetadataService.Protocol + "://" +
+				config.Conf.MetadataService.Host + ":" +
+				config.Conf.MetadataService.Port
+
+			mdc := metadata.NewDeviceClient(
+				urlclient.New(
+					ctx,
+					clients.CoreMetaDataServiceKey,
+					clients.ApiDeviceRoute,
+					15000,
+					url +  clients.ApiDeviceRoute,
+				),
+			)
+
+			//devices, err := mdc.Devices(ctx)
+
+			err := mdc.DeleteByName(ctx, deviceID)
 
 			if err != nil {
-				fmt.Println(err)
+				fmt.Printf("Removed: %s\n", deviceID)
```

Comment:
Is this not a delete failure case? Believe you wanted to test for == nil

Reply:
You're right. it should be `err == nil`. We're checking if `DeleteByName` was successful. If not, we try deleting by ID.